crates/polars-ops/src/series/ops/index_of.rs

```diff
@@ -0,0 +1,168 @@
+use arrow::array::{Array, BinaryArray, PrimitiveArray};
+use arrow::types::NativeType;
+use polars_core::downcast_as_macro_arg_physical;
+use polars_core::prelude::*;
+use polars_utils::total_ord::TotalEq;
+use row_encode::encode_rows_unordered;
+
+/// Annoyingly, iter() and values_iter() on Arrow arrays are not part of the
+/// Array trait, so we have to create our own.
+trait IterableArray<'a>: Array {
```

Comment:
We shouldn't need a new trait here, `iter()` and `values_iter()` should already be available through the `StaticArray` trait